### PR TITLE
new playwright web CI pipeline

### PR DESCRIPTION
**Pull request by Vere-Grey**
[#15605](https://github.com/trezor/trezor-suite/issues/15605)
This is one of the first steps of moveing from Cypress to playwright. This PR adds new GHA pipeline that runs the web tests against deployed PR web instance.

Finishes @HajekOndrej draft of PW web pipeline.
Conversion of autocomplete test file from CY to PW

### CODE REVIEW

File: packages/suite-desktop-core/e2e/support/fixtures.ts [hunk 1]
```diff
@@ -37,24 +38,42 @@ const test = base.extend<Fixtures>({
...
+        if (appContext) {
+            const window = await appContext.firstWindow();
+            await window.context().tracing.start({ screenshots: true, snapshots: true });
+            await use(window);
+            const tracePath = `${testInfo.outputDir}/trace.electron.zip`;
+            await window.context().tracing.stop({ path: tracePath });
+        } else {
+            await page.goto('./');
+            await use(page);
+        }
```
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/support/fixtures.ts:76]: combination of web/electron and starting or not the emulator clatters these fixtures a bit. I imagine as we convert more and more tests, we will have better understanding what we need and how. Then we should refactor this crucial part to be a cleaner and simpler

**Review by HajekOndrej — APPROVED** [review]

File: packages/suite-desktop-core/package.json [hunk 2]
```diff
@@ -10,7 +10,9 @@
         "build:core": "yarn g:rimraf dist && TS_NODE_PROJECT=\"tsconfig.json\" yarn webpack --config ./webpack/core.webpack.config.ts",
         "type-check": "yarn g:tsc --build tsconfig.json",
         "test:unit": "yarn g:jest",
-        "test:e2e": "yarn xvfb-maybe -- playwright test --config=./e2e/playwright.config.ts"
+        "test:e2e": "yarn xvfb-maybe -- playwright test --config=./e2e/playwright.config.ts",
```
Comment (HajekOndrej) [inline packages/suite-desktop-core/package.json:13]: I think we can remove this line

**Review by vdovhanych — APPROVED** [review]
For the CI part.

File: packages/suite-desktop-core/e2e/support/fixtures.ts [hunk 3]
```diff
@@ -37,24 +38,42 @@ const test = base.extend<Fixtures>({
...
         // We need to ensure emulator is running before launching the suite
         if (startEmulator) {
             await trezorUserEnvLink.stopBridge();
+            await trezorUserEnvLink.stopEmu();
             await trezorUserEnvLink.connect();
             await trezorUserEnvLink.startEmu({ wipe: true });
             await trezorUserEnvLink.setupEmu(emulatorConf);
         }
-        const suite = await launchSuite();
-        await use(suite.electronApp);
```
Comment (mroz22) [inline packages/suite-desktop-core/e2e/support/fixtures.ts:49]: what is this `use` thingy? I don't even remember this code

File: packages/suite-desktop-core/e2e/support/pageActions/settingsActions.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { BackendType, NetworkSymbol } from '@suite-common/wallet-config';
 
 export class SettingsActions {
     private readonly window: Page;
-    private readonly TIMES_CLICK_TO_SET_DEBUG_MODE = 5;
+    private readonly TIMES_CLICK_TO_SET_DEBUG_MODE = 7;
```
Comment (mroz22) [inline packages/suite-desktop-core/e2e/support/pageActions/settingsActions.ts:7]: this has changed?

File: packages/suite-desktop-core/e2e/tests/general/suite-guide.test.ts [hunk 5]
```diff
@@ -1,34 +1,36 @@
...
- * 4. Submit bug report (reporttext)
- */
-test('Send a bug report', async ({ suiteGuidePage }) => {
-    await suiteGuidePage.openPanel();
-    await suiteGuidePage.supportAndFeedbackButton.click();
-    await suiteGuidePage.sendBugReport({
-        location: 'account',
-        report: 'Henlo this is testy test writing hangry test user report',
+test.describe('Suite Guide', { tag: '@suite' }, () => {
+    test.use({ startEmulator: false });
```
Comment (mroz22) [inline packages/suite-desktop-core/e2e/tests/general/suite-guide.test.ts:4]: aha, this is some kind of beforeeach

**Review by mroz22 — APPROVED** [review]
I did not read it very carefully but it looks like it is going the right direction 👍

File: packages/suite-desktop-core/e2e/support/pageActions/settingsActions.ts (hunk 4 above)
Comment (Vere-Grey) [reply packages/suite-desktop-core/e2e/support/pageActions/settingsActions.ts:7]: It is still 5.
But I have encountered a test run where 5 clicks failed. My assumption is the page probably was not fully loaded and the counter got reset once it finished. So either we find the event that did that and try to check for loaded page or we can keep a this extra buffer of clicks.

File: packages/suite-desktop-core/e2e/support/fixtures.ts (hunk 3 above)
Comment (Vere-Grey) [reply packages/suite-desktop-core/e2e/support/fixtures.ts:49]: That is playwright function that allow you to provide fixtures to tests or other fixtures. The fixture can be any asset that you prepare, either page object model class or extension of default page object with your own custom methods

So in test for example. you can import two already initilized fixtures `onboardingPage`, `settingsPage` and do whatever you need. Keeping the test clearer and more easy to read. 
```
    test.beforeEach(async ({ onboardingPage, settingsPage }) => {
        await onboardingPage.disableFirmwareHashCheck();
        await onboardingPage.completeOnboarding();
        await settingsPage.navigateTo();
        await settingsPage.deviceTabButton.click();
    });
```

File: packages/suite-desktop-core/e2e/tests/general/suite-guide.test.ts (hunk 5 above)
Comment (Vere-Grey) [reply packages/suite-desktop-core/e2e/tests/general/suite-guide.test.ts:4]: Yeah, kind of. `test.use` allow you to configure board range of test properties. Starting from browser type, baseUrl etc. And thanks to our change even how you setup or whether to setup the emulator.

Majority of our test need emulator, and need to get thru the initial Onboarding. We removed the need defining that in every test by moving it to fixtures. Also we are handling there logic web vs desktop and specific needs for the emulator/bridges.
But the test still need to specify params for the emulator/bridges. We handle that thru test.use
I will see how practical it is in large scale and improve based on that. Also any feedback is welcomed.

**Pull request merged**